### PR TITLE
Add the option to choose additional scopes at sign-in

### PR DESCRIPTION
**Pull request by AntonKozlovskyi**

### CODE REVIEW

**Comment by artemHrebinikChisw** [comment]
Very useful functionality. Today I ran into this problem on the project.
How long does it usually take to review such PR?

**Comment by petea** [comment]
Thanks for the PR.  Please note that the separation of `signIn` and `addScopes` is intentional with this release, as mentioned in #23.  Feel free to discuss relevant use cases and impact.

**Comment by ivopintodasilva** [comment]
Any idea on the timeline to conclude if this change will be approved or not?
This is preventing us from updating GoogleSignIn to the latest version...

**Comment by dpanzer** [comment]
@petea It seems that the use cases for this functionality have been documented in #23. I would add my predicament as an SDK developer is that I don't know whether to release an update with a breaking change based on the new 2-step add scope flow, or whether this PR will get approved and I can release an update where the upgrade to v6 will be transparent to users of my SDK (obviously strongly prefer the latter, but overall just want to know what the verdict will be). Thanks!

**Comment by reslear** [comment]
any news?

**Comment by foxware00** [comment]
@petea To weigh in, it's also paramount for us that we don't prompt twice give the first screen of our app uses the calendar permission. I think a lot of people have outlined their concerns and use cases in the issue below. It provides a poor user experience and also means we won't be upgrading to this library in the meantime.

I'm not entirely sure this library should be opinonated on UX flow. If it's required that apps request scope at time of using that feature, surely that is up to the individual use cases. Those apps which require scope extensions immediately are penalised.

**Comment by petea** [comment]
@AntonKozlovskyi thanks again for your contribution!  We now plan to ship this with the 6.2 release.  See my [update](https://github.com/google/GoogleSignIn-iOS/issues/23#issuecomment-1061230684) on #23.

**Review by petea — APPROVED** [review]

File: GoogleSignIn/Sources/GIDSignIn.m [hunk 1]
```diff
@@ -192,6 +192,20 @@ - (void)signInWithConfiguration:(GIDConfiguration *)configuration
   [self signInWithOptions:options];
 }
 
+- (void)signInWithConfiguration:(GIDConfiguration *)configuration
```
Comment (etrdotcom) [inline GoogleSignIn/Sources/GIDSignIn.m:220]: 000IOEC. P kikx.ios@yahoo.c omreceipt.here

**Pull request merged**